### PR TITLE
Amazon Linux 2023 support

### PR DESCRIPTION
**Pull request by sbezugliy**
### Proposed changes

`Amazon Linux 2023` is not supported by current Ansible role. Also it has some significant differences with `Amazon Linux 1, 2`, so same actions aren't compatible. List of differences and issues:
  - `Fedora` based. (Not sure about full support of Fedora repositories)
  - Doesn't support some packages from `CentOS`, `RHEL` and other mainline distributive repositories, have unsolvable dependencies graphs.
  - Doesn't support `EPEL` repository. `EPEL` is not recommended by creators of distributive.
  - `Dnf` is default package manager. No yum installed.
  - `Python 3` is used by default. `Yum` not supported and require some additional dependencies for Python or installing `python ~= 2.7`.
    Mismatch of versions between `Amazon Linux 2023 Repository` and official `Nginx repository` for Amazon Linux 2023. Added both Nginx repositories: `stable` and `mainline`. Mismatch in versioning convention for Nginx: amazon linux repository uses `nginx-1:<major_version>.<minor_version>`, when Nginx official repository uses `nginx:<major_version>.<minor_version>`. So, as example: `nginx-1:1.20.4` is recognized as higher version then `nginx:1.23.1`. The same with module packages.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/ansible-role-nginx/blob/main/CONTRIBUTING.md) document
- [x] I have added Molecule tests that prove my fix is effective or that my feature works
- [x] I have checked that any relevant Molecule tests pass after adding my changes
- [x] I have updated any relevant documentation ([`defaults/main/*.yml`](https://github.com/nginxinc/ansible-role-nginx/blob/main/defaults/main/), [`README.md`](https://github.com/nginxinc/ansible-role-nginx/blob/main/README.md) and [`CHANGELOG.md`](https://github.com/nginxinc/ansible-role-nginx/blob/main/CHANGELOG.md))

### CODE REVIEW

**Comment by alessfg** [comment]
Thanks for the PR! I'll get to it asap!

Two initial thoughts before I actually do a review:

a) Looks like you ran `ansible-lint --write`? I might ask you to revert some of the changes the tool does by default. The changes it does are sometimes not that great for readibility
b) Looks like some tests are failing? I had a quick look and all Molecule distro specs are identical across all scenarios, so I am not entirely sure why it sometimes fails? A simple rerun might be good enough to fix those errors, but if they keep failing we might need to look into it a bit more.

**Review by jskelly84 — APPROVED** [review]

**Comment by alessfg** [comment]
Ok some of these tests are now failing because a new version of NGINX got released over night and it broke some of the Molecule verification tests. I'll open a PR with a fix for that either later today or tomorrow and then we can try to run these tests again :)

File: defaults/main/amplify.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@
 # Requires access to either the NGINX stub_status or the NGINX Plus REST API.
 # Default is null.
 nginx_amplify_enable: false
-nginx_amplify_api_key: null
+nginx_amplify_api_key:
```
Comment (alessfg) [inline defaults/main/amplify.yml:7]: Needs to be changed back to null (for clarity)

File: meta/main.yml [hunk 2]
```diff
@@ -17,13 +17,13 @@ galaxy_info:
     - name: Debian
       versions: [bullseye]
     - name: EL
-      versions: ['7', '8', '9']
+      versions: ["7", "8", "9"]
```
Comment (alessfg) [inline meta/main.yml:20]: Can you change double quotes here back to single quotes (I use single quotes for array values)? (Same for the other array/list values in the metadata.)

File: molecule/source/molecule.yml
Comment (alessfg) [inline molecule/source/molecule.yml:1]: Amazon Linux 2023 tests are missing from the default/distribution/downgrade/downgrade-plus/plus Molecule scenarios :)

File: molecule/version/converge.yml [hunk 3]
```diff
@@ -15,7 +15,8 @@
     - name: Set repo if Red Hat
       ansible.builtin.set_fact:
         ngx_version: -1.23.2-1.{{ (ansible_facts['distribution'] == "Amazon") | ternary('amzn2', ('el' + ansible_facts['distribution_major_version'] | string)) }}.ngx
-        njs_version: -1.23.2+0.7.9-1.{{ (ansible_facts['distribution'] == "Amazon") | ternary('amzn2', ('el' + ansible_facts['distribution_major_version'] | string)) }}.ngx
+        njs_version: -1.23.2+0.7.9-1.{{ (ansible_facts['distribution'] == "Amazon") | ternary('amzn2', ('el' + ansible_facts['distribution_major_version'] | string))
+          }}.ngx
```
Comment (alessfg) [inline molecule/version/converge.yml:19]: This needs to go back to how it was before. A single line is much preferable to what it looks like now.

File: tasks/amplify/setup-debian.yml [hunk 4]
```diff
@@ -2,6 +2,7 @@
 - name: (Debian/Ubuntu) Add NGINX Amplify agent repository
   ansible.builtin.apt_repository:
     filename: nginx-amplify
-    repo: deb [arch=amd64 signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://packages.amplify.nginx.com/py3/{{ ansible_facts['distribution'] | lower }}/{{ ansible_facts['distribution_release'] | lower }} amplify-agent
+    repo: deb [arch=amd64 signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://packages.amplify.nginx.com/py3/{{ ansible_facts['distribution'] | lower
```
Comment (alessfg) [inline tasks/amplify/setup-debian.yml:5]: Needs to go back to a single line :)

File: tasks/amplify/setup-redhat.yml [hunk 5]
```diff
@@ -2,7 +2,8 @@
 - name: (Amazon Linux/CentOS/RHEL) Add NGINX Amplify agent repository
   ansible.builtin.yum_repository:
     name: nginx-amplify
-    baseurl: https://packages.amplify.nginx.com/{{ (ansible_facts['distribution_major_version'] is version('7', '==')) | ternary('', 'py3/') }}{{ (ansible_facts['distribution'] == 'Amazon') | ternary('amzn', 'centos') }}/$releasever/$basearch/
+    baseurl: https://packages.amplify.nginx.com/{{ (ansible_facts['distribution_major_version'] is version('7', '==')) | ternary('', 'py3/') }}{{ (ansible_facts['distribution']
+      == 'Amazon') | ternary('amzn', 'centos') }}/$releasever/$basearch/
```
Comment (alessfg) [inline tasks/amplify/setup-redhat.yml:6]: Needs to go back to a single line :)

File: tasks/config/setup-logrotate.yml [hunk 6]
```diff
@@ -11,10 +11,25 @@
   when: ansible_facts['os_family'] == 'Debian'
 
 - name: (Amazon Linux/CentOS/Oracle Linux/RHEL) Install logrotate
+  vars:
```
Comment (alessfg) [inline tasks/config/setup-logrotate.yml:14]: I am not entirely sure we need this vars block? We can check whether it's Amazon Linux 2023 in the `when` block directly.

File: tasks/config/setup-logrotate.yml [hunk 7]
```diff
@@ -11,10 +11,25 @@
...
   ansible.builtin.yum:
     name: logrotate
     state: present
-  when: ansible_facts['os_family'] == 'RedHat'
+  when:
+    - ansible_facts['os_family'] == 'RedHat'
+    - distribution != 'Amazon Linux 2023'
+
+- name: (Amazon Linux 2023) Install logrotate
+  vars:
```
Comment (alessfg) [inline tasks/config/setup-logrotate.yml:24]: As before, we can check if whether it's Amazon Linux 2023 in the when block directly.

File: tasks/main.yml [hunk 8]
```diff
@@ -10,8 +10,7 @@
 
 - name: Set up signing keys
   ansible.builtin.include_tasks: "{{ role_path }}/tasks/keys/setup-keys.yml"
-  when: (nginx_enable | bool and nginx_install_from == 'nginx_repository')
-        or nginx_amplify_enable | bool
+  when: (nginx_enable | bool and nginx_install_from == 'nginx_repository') or nginx_amplify_enable | bool
```
Comment (alessfg) [inline tasks/main.yml:13]: Can we split this back into two lines? It was split on purpose to help with clarity.

File: tasks/config/setup-logrotate.yml [hunk 9]
```diff
@@ -11,10 +11,25 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
+  ansible.builtin.dnf:
+    enablerepo: nginx-mainline-repo
```
Comment (alessfg) [inline tasks/config/setup-logrotate.yml:27]: Do we need to set enablerepo/disablerepo even if we are not installing NGINX in this task?

File: tasks/modules/install-modules.yml [hunk 10]
```diff
@@ -8,9 +8,11 @@
     - nginx_install_epel_release | bool
 
 - name: (Amazon Linux/RHEL 7) Install GeoIP2 dependency
+  vars:
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:11]: See comment on logrotate tasks. Let's do the comparison for this directly in the `when` conditional block.

File: tasks/modules/install-modules.yml [hunk 11]
```diff
@@ -8,9 +8,11 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   when:
-    - (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] is version('7', '=='))
-      or ansible_facts['distribution'] == "Amazon"
+    - (ansible_facts['distribution'] == "RedHat" and ansible_facts['distribution_major_version'] is version('7', '==')) or ansible_facts['distribution'] == "Amazon
+      and distribution != 'Amazon LInux 2023'"
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:15]: While at it, if we are going to separate this conditional into two lines, let's do it at the `or`, not like it is done right now.

File: tasks/modules/install-modules.yml [hunk 12]
```diff
@@ -23,20 +25,36 @@
         name: https://rpmfind.net/linux/centos/7.9.2009/os/x86_64/Packages/libmaxminddb-1.2.0-6.el7.x86_64.rpm
       when: "'libmaxminddb' not in packages['results']"
 
+- name: (Amazon Linux 2023) Install GeoIP2 dependency
+  vars:
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:29]: See all my previous comments re vars block vs conditionals.

File: tasks/modules/install-modules.yml [hunk 13]
```diff
@@ -23,20 +25,36 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
+  ansible.builtin.dnf:
+    enablerepo: nginx-mainline-repo
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:32]: Do we need to enablerepo/disablerepo here?

File: tasks/modules/install-modules.yml [hunk 14]
```diff
@@ -23,20 +25,36 @@
...
+    disablerepo: amazonlinux
+    name: libmaxminddb
+  when:
+    - distribution == "Amazon Linux 2023"
+    - '"geoip2" in nginx_modules'
+
 - name: (AlmaLinux/Amazon Linux/Oracle Linux/RHEL/Rocky Linux) Install GeoIP2 and/or OpenTracing EPEL dependency
+  vars:
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:40]: Remove :)

File: tasks/modules/install-modules.yml [hunk 15]
```diff
@@ -23,20 +25,36 @@
...
+    - distribution != "Amazon Linux 2023"
     - ('opentracing' in nginx_modules)
     - nginx_install_epel_release | bool
   block:
     - name: (AlmaLinux/Amazon Linux/Oracle Linux/RHEL/Rocky Linux) Import EPEL GPG key
       ansible.builtin.rpm_key:
         state: present
-        key: https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version'])) }}
+        key: https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version']))
+          }}
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:52]: Combine into single line.

File: tasks/modules/install-modules.yml [hunk 16]
```diff
@@ -23,20 +25,36 @@
...
 
     - name: (AlmaLinux/Amazon Linux/Oracle Linux/RHEL/Rocky Linux) Install EPEL
       ansible.builtin.yum:
-        name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version'])) }}.noarch.rpm
+        name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version']))
+          }}.noarch.rpm
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:57]: Combine into single line.

File: tasks/modules/install-modules.yml [hunk 17]
```diff
@@ -49,21 +67,18 @@
 
 - name: Install NGINX modules
   ansible.builtin.package:
-    name: "nginx-{{ (nginx_type == 'plus') | ternary('plus-', '') }}module-{{ item['name'] | default(item) }}\
-          {{ (nginx_repository is not defined and ansible_facts['os_family'] == 'Alpine' and nginx_type != 'plus') | ternary('@nginx', '') }}{{ item['version'] | default('') }}"
+    name: nginx-{{ (nginx_type == 'plus') | ternary('plus-', '') }}module-{{ item['name'] | default(item) }}{{ (nginx_repository is not defined and ansible_facts['os_family']
+      == 'Alpine' and nginx_type != 'plus') | ternary('@nginx', '') }}{{ item['version'] | default('') }}
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:71]: Combine to previous format.

File: tasks/modules/install-modules.yml [hunk 18]
```diff
@@ -49,21 +67,18 @@
...
+    - (item['name'] | default(item) in nginx_modules_list and nginx_type == 'opensource') or (item['name'] | default(item) in nginx_plus_modules_list and nginx_type
+      == 'plus')
+    - not (item['name'] | default(item) == 'brotli') or not (ansible_facts['os_family'] == 'Suse' and ansible_facts['distribution_major_version'] is version('12',
+      '=='))
+    - not (item['name'] | default(item) == "geoip") or not ((ansible_facts['os_family'] == 'FreeBSD') or (ansible_facts['os_family'] == 'RedHat' and ansible_facts['distribution_major_version']
+      is version('8', '>=')))
+    - not (item['name'] | default(item) == 'geoip2') or not (ansible_facts['os_family'] == 'Suse')
+    - not (item['name'] | default(item) == 'lua') or not (ansible_facts['architecture'] == 's390x')
+    - not (item['name'] | default(item) == 'opentracing') or not (ansible_facts['os_family'] == 'Suse' and ansible_facts['distribution_major_version'] is version('12',
+      '=='))
```
Comment (alessfg) [inline tasks/modules/install-modules.yml:84]: Revert to previous format :)

File: tasks/opensource/install-distribution.yml [hunk 19]
```diff
@@ -8,11 +8,13 @@
     - name: (CentOS/RHEL 7) Import the EPEL repository GPG key
       ansible.builtin.rpm_key:
         state: present
-        key: https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version'])) }}
+        key: https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version']))
+          }}
```
Comment (alessfg) [inline tasks/opensource/install-distribution.yml:12]: Revert to previous format :)

File: tasks/opensource/install-distribution.yml [hunk 20]
```diff
@@ -8,11 +8,13 @@
...
 
     - name: (CentOS/RHEL 7) Install the EPEL repository
       ansible.builtin.yum:
-        name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version'])) }}.noarch.rpm
+        name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ((ansible_facts['distribution'] == 'Amazon') | ternary('7', ansible_facts['distribution_major_version']))
+          }}.noarch.rpm
```
Comment (alessfg) [inline tasks/opensource/install-distribution.yml:17]: Revert to previous format :)

File: tasks/opensource/install-distribution.yml [hunk 21]
```diff
@@ -32,13 +34,35 @@
     - ansible_facts['distribution_major_version'] is version('7', '==')
 
 - name: (Amazon Linux) Enable the NGINX Amazon extras package repository
+  vars:
```
Comment (alessfg) [inline tasks/opensource/install-distribution.yml:37]: Remove :)

File: tasks/opensource/install-distribution.yml [hunk 22]
```diff
@@ -32,13 +34,35 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
+  ansible.builtin.dnf:
+    name: nginx{{ nginx_version | default('') }}
+    disablerepo: amazonlinux
+    enablerepo: nginx-mainline-repo
+    state: "{{ nginx_state }}"
+  notify: (Handler) Run NGINX
+  when:
+    - ansible_facts['distribution'] == 'Amazon'
+    - distribution == 'Amazon Linux 2023'
```
Comment (alessfg) [inline tasks/opensource/install-distribution.yml:68]: This task involves installing NGINX from the Amazon repo. I am pretty sure that's not what this block does? I am also not entirely sure if we even need an Amazon Linux 2023 specific task or whether NGINX is available in the core  Amazon Linux 2023 packages? (Judging by the fact you had to disable the amazonlinux repo I assume it is?)

File: tasks/opensource/install-redhat.yml [hunk 23]
```diff
@@ -1,26 +1,82 @@
 ---
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ (nginx_setup == 'uninstall') | ternary('Remove', 'Configure') }} NGINX repository
+  vars:
```
Comment (alessfg) [inline tasks/opensource/install-redhat.yml:3]: Remove :)

File: tasks/opensource/install-redhat.yml [hunk 24]
```diff
@@ -1,26 +1,82 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.yum_repository:
     name: nginx
-    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat')))) }}"
+    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat'))))\
+      \ }}"
```
Comment (alessfg) [inline tasks/opensource/install-redhat.yml:8]: Revert :)

File: tasks/opensource/install-redhat.yml [hunk 25]
```diff
@@ -1,26 +1,82 @@
...
     gpgcheck: true
     mode: "0644"
     module_hotfixes: true
     state: "{{ (nginx_state == 'uninstall') | ternary('absent', 'present') }}"
-  when: nginx_manage_repo | bool
+  when:
+    - nginx_manage_repo | bool
+    - distribution != 'Amazon Linux 2023'
+
+- name: (Amazon Linux 2023) {{ (nginx_setup == 'uninstall') | ternary('Remove', 'Configure') }} NGINX stable repository
```
Comment (alessfg) [inline tasks/opensource/install-redhat.yml:19]: Check what's being done behind the scenes in the vars file. I'd like to keep everything contained within the task that already exists.

File: tasks/opensource/install-redhat.yml [hunk 26]
```diff
@@ -1,26 +1,82 @@
...
+    - nginx_manage_repo | bool
+    - distribution == 'Amazon Linux 2023'
 
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) Force Yum cache refresh
   ansible.builtin.command: yum clean metadata
   when: nginx_manage_repo | bool
   changed_when: false
 
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ nginx_setup | capitalize }} NGINX
+  vars:
```
Comment (alessfg) [inline tasks/opensource/install-redhat.yml:59]: Remove :)

File: tasks/opensource/install-source.yml [hunk 27]
```diff
@@ -117,10 +141,23 @@
       when: ansible_facts['os_family'] == 'Alpine'
 
     - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) Install PCRE dependency from package
+      vars:
```
Comment (alessfg) [inline tasks/opensource/install-source.yml:144]: Remove :)

File: tasks/opensource/install-source.yml [hunk 28]
```diff
@@ -48,6 +48,8 @@
           when: ansible_facts['distribution_major_version'] is version('8', '==')
 
         - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) Install build tools
+          vars:
```
Comment (alessfg) [inline tasks/opensource/install-source.yml:51]: Remove :)

File: tasks/opensource/install-source.yml [hunk 29]
```diff
@@ -142,7 +179,8 @@
   block:
     - name: Download PCRE dependency
       ansible.builtin.get_url:
-        url: "{{ (pcre_release == 2) | ternary('https://github.com/PCRE2Project/pcre2/releases/download/pcre2-' ~ pcre_version ~ '/pcre2-' ~ pcre_version ~ '.tar.gz', 'https://ftp.exim.org/pub/pcre/pcre-' ~ pcre_version ~ '.tar.gz') }}"
+        url: "{{ (pcre_release == 2) | ternary('https://github.com/PCRE2Project/pcre2/releases/download/pcre2-' ~ pcre_version ~ '/pcre2-' ~ pcre_version ~ '.tar.gz',\
+          \ 'https://ftp.exim.org/pub/pcre/pcre-' ~ pcre_version ~ '.tar.gz') }}"
```
Comment (alessfg) [inline tasks/opensource/install-source.yml:183]: Revert :)

File: tasks/opensource/install-source.yml [hunk 30]
```diff
@@ -186,10 +223,23 @@
       when: ansible_facts['os_family'] == 'Alpine'
 
     - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) Install ZLib dependency from package
+      vars:
```
Comment (alessfg) [inline tasks/opensource/install-source.yml:226]: Remove :)

File: tasks/opensource/install-source.yml [hunk 31]
```diff
@@ -298,7 +360,6 @@
         dest: /tmp/openssl-{{ openssl_version }}
         mode: "0700"
         extra_opts: [--strip-components=1]
-
```
Comment (alessfg) [inline tasks/opensource/install-source.yml:301]: Not entirely sure why Ansible lint removed this line but let's put it back :)

File: tasks/plus/install-redhat.yml [hunk 32]
```diff
@@ -1,28 +1,63 @@
 ---
-- name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ (nginx_license_status is defined or nginx_setup == 'uninstall') | ternary('Remove', 'Configure') }} NGINX Plus repository
+- name: (AlmaLinux/Amazon Linux/Amazon Linux 2023/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ (nginx_license_status is defined or nginx_setup == 'uninstall') | ternary('Remove',
+    'Configure') }} NGINX Plus repository
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:3]: Revert back to a single line :)

File: tasks/plus/install-redhat.yml [hunk 33]
```diff
@@ -1,28 +1,63 @@
...
   ansible.builtin.yum_repository:
     name: nginx-plus
-    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_plus_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat')))) }}"
+    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_plus_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat'))))\
+      \ }}"
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:7]: Revert :)

File: tasks/plus/install-redhat.yml [hunk 34]
```diff
@@ -1,28 +1,63 @@
...
     description: NGINX Plus Repository
     sslclientcert: /etc/ssl/nginx/nginx-repo.crt
     sslclientkey: /etc/ssl/nginx/nginx-repo.key
     enabled: true
     gpgcheck: true
     state: "{{ nginx_license_status | default((nginx_setup == 'uninstall') | ternary('absent', 'present')) }}"
     mode: "0644"
-  when: nginx_manage_repo | bool
+  when:
+    - nginx_manage_repo | bool
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:16]: Revert :)

File: tasks/plus/install-redhat.yml [hunk 35]
```diff
@@ -1,28 +1,63 @@
...
 
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) Force Yum cache refresh
+  vars:
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:19]: Remove :)

File: tasks/plus/install-redhat.yml [hunk 36]
```diff
@@ -1,28 +1,63 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.command: yum clean metadata
-  when: nginx_manage_repo | bool
+  when:
+    - nginx_manage_repo | bool
+    - distribution != 'Amazon Linux 2023'
+  changed_when: false
+
+- name: (Amazon Linux 2023) Force Yum cache refresh
+  vars:
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:28]: Remove :)

File: tasks/plus/install-redhat.yml [hunk 37]
```diff
@@ -1,28 +1,63 @@
...
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
+  ansible.builtin.command: dnf clean metadata
+  when:
+    - nginx_manage_repo | bool
+    - distribution == 'Amazon Linux 2023'
   changed_when: false
 
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ nginx_setup | capitalize }} NGINX Plus
+  vars:
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:37]: Remove :)

File: tasks/plus/install-redhat.yml [hunk 38]
```diff
@@ -1,28 +1,63 @@
...
     allow_downgrade: "{{ omit if ansible_version['full'] is version('2.12', '<') else true }}"
   ignore_errors: "{{ ansible_check_mode }}"
-  when: nginx_license_status is not defined
+  when:
+    - nginx_license_status is not defined
+    - distribution != 'Amazon Linux 2023'
+  notify: (Handler) Run NGINX
+
+- name: (Amazon Linux 2023) {{ nginx_setup | capitalize }} NGINX Plus
+  vars:
```
Comment (alessfg) [inline tasks/plus/install-redhat.yml:51]: Remove :)

File: tasks/prerequisites/install-dependencies.yml [hunk 39]
```diff
@@ -14,11 +14,26 @@
   when: ansible_facts['os_family'] == 'Debian'
 
 - name: (Amazon Linux/CentOS/Oracle Linux/RHEL) Install dependencies
+  vars:
```
Comment (alessfg) [inline tasks/prerequisites/install-dependencies.yml:17]: Remove :)

File: tasks/prerequisites/install-dependencies.yml [hunk 40]
```diff
@@ -14,11 +14,26 @@
...
     name: "{{ nginx_redhat_dependencies }}"
     update_cache: true
     state: latest # noqa package-latest
-  when: ansible_facts['os_family'] == 'RedHat'
+  when:
+    - ansible_facts['os_family'] == 'RedHat'
+    - distribution != 'Amazon Linux 2023'
+
+- name: (Amazon Linux 2023) Install dependencies
+  vars:
```
Comment (alessfg) [inline tasks/prerequisites/install-dependencies.yml:28]: Remove :)

File: tasks/prerequisites/prerequisites.yml [hunk 41]
```diff
@@ -17,3 +17,20 @@
     - name: Configure SELinux
       ansible.builtin.include_tasks: "{{ role_path }}/tasks/prerequisites/setup-selinux.yml"
       when: ansible_facts['selinux']['mode'] is defined
+
+- name: (Amazon Linux 2023) Set up SELinux
+  vars:
```
Comment (alessfg) [inline tasks/prerequisites/prerequisites.yml:22]: Remove :)

File: tasks/prerequisites/setup-selinux.yml [hunk 42]
```diff
@@ -1,11 +1,23 @@
 ---
 - name: (CentOS/RHEL) Install dependencies
+  vars:
```
Comment (alessfg) [inline tasks/prerequisites/setup-selinux.yml:3]: Remove :)

File: tasks/validate/validate.yml
Comment (alessfg) [inline tasks/validate/validate.yml:1]: Revert all changes here to the previous format

File: vars/main.yml
Comment (alessfg) [inline vars/main.yml:1]: Revert all formatting changes here

**Review by alessfg — CHANGES_REQUESTED** [review]
Thanks for the PR! It's a good first pass, but we need to:
* Revert all the Ansible Lint formatting changes (the auto formatting capabilities are very much a WIP)
* Add some missing Molecule tests (and figure out why the ones that are here are failing?)
* Remove your usage of `vars` blocks throughout multiple tasks. It adds extra code when a simple double conditional should be enough (and would follow the same format used throughout the rest of the role).
* Double check some `dnf` installation tasks (I made some comments where relevant).

P.S.: I started writing out long/detailed comments but at some stage started making smaller one word comments. LMK if you have any questions 😄

**Review by jskelly84 — APPROVED** [review]

**Comment by alessfg** [comment]
Thanks for the updates!

1. Looks like some Molecule scenarios are still missing 😄 (e.g. there is no AL2023 platform in the `default` scenario). If you could comment out/remove the `platform` parameter (e.g. https://github.com/nginxinc/ansible-role-nginx/pull/597/files#diff-f653be9e6e7813c18a248edeb270c6a8034e1ed3b0ea3e49e6e6a9192f740201R65) for the time being, that'd be great. The pipeline has been running into some emulation related issues.
2. The linter is failing -- not entirely sure why but there might be some issues with the YAML code you've written.

**Comment by sbezugliy** [comment]
Molecule is new for me, but I know what to do with it. So, trying...
Yes, I'll check one more time syntax and next fix. Looks like typo with quotes.

**Comment by alessfg** [comment]
Closing this PR in favor of #706.

**Pull request closed without merging**